### DISCUSSION TITLE
Keyball 39の特定のキー反応不良について

### DISCUSSION THREAD

**Discussion by y-y-yossy** [Yowkees/keyball#510] [Q&A]
お世話になります。

当方、初めての自作キーボードかつハードウェアの知識に乏しいため、
以下についてご教示いただけますと幸いです...。
keyball39、USB-C(ProMicro)、右手トラックボールです。

[手順書](https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp.md)を参考にさせていただき、トラックボール設置の手前までの作業が完了しているところです。

ファームウェアの書き込みが完了し、入力のテストをしていたのですが、
右側、左側ともに1部のキーが反応致しません。

■ 反応しないキー
・右側：最も左下のキー（親指で操作する2つのキーのうち左(外)側）
・左側：ProMicroから数えて３列目４行目のキー

■ 試してみたこと
・ProMicroの左右交換 → 交換前と同じキーが反応しない
・キースイッチの交換 → 改善せず
・ダイオードの交換 → 改善せず
・キーソケットの交換 → 改善せず
・はんだ付けのやり直し → 改善せず
・キースイッチを使用せずにピンセットでの通電 → 反応せず
・Promicroのショートがないか確認 → 改善せず

■ 写真
[Gドライブ](https://drive.google.com/drive/folders/1Mr8EXI6eTvLatQIfIkwPpUCTXif6qOU-?usp=sharing)

恐れ入りますが、ご教示のほど何卒よろしくお願いいたします。

**Comment by y-y-yossy** [comment]
Thank you for your help.

This is my first self-made keyboard and I don't have much knowledge about hardware, so
I would appreciate it if you could tell me about the following...
keyball39, USB-C (ProMicro), right hand trackball.

Using the procedure manual as a reference, we have completed the work up to the point of installing the trackball.

I completed writing the firmware and was testing the input.
One key on both the right and left side does not respond.

■ Keys that do not respond
・Right side: Lower left key (the left (outer) side of the two keys operated with the thumb)
・Left side: Key in the 3rd column and 4th row counting from ProMicro

■ What I tried
・Right and left replacement of ProMicro → Same keys as before replacement do not respond
・Replacement of key switch → No improvement
・Replacement of diode → No improvement
・Replacement of key socket → No improvement
・Redo soldering → No improvement
・Power on with tweezers without using a key switch → No response
・Check if there is a short circuit in Promicro → No improvement

■ Photos
G drive: https://drive.google.com/drive/folders/1Mr8EXI6eTvLatQIfIkwPpUCTXif6qOU-?usp=sharing

We apologize for the inconvenience, and thank you for your understanding.

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

様々なご対応をしていただいて改善が見られないということで、もしかしたら何かの拍子に基板の配線が傷つき導通に影響が出てしまったかもしれません。
添付させていただいた写真の赤点どうしの導通を確認していただけますでしょうか。

![Right_01](https://github.com/Yowkees/keyball/assets/143698349/0fb793d4-a263-4dc5-aeb7-de77d4eaba6b)
![Left_01](https://github.com/Yowkees/keyball/assets/143698349/34942d02-7a08-4f64-a8ca-bbe8ed6477d1)

**Reply by y-y-yossy** [reply]
@Ryo624 
ご回答ありがとうございます。

ご指示いただいた内容通りに通電してみましたがどちらもOLEDの表示は何も変わらず、PCにも何も出力されませんでした...。
参考になるか分かりませんが、以下の事象も確認できましたのでご共有いたします。

【事象1】
右デバイスで、問題のキーの赤点とCOL3右の穴（はんだ付けしていない部分）を通電すると「Key: R7 C3 K91」がOLEDに表示され、PCには"日本語/英語の変換"のキーとして入力されました。
![IMG_1702](https://github.com/Yowkees/keyball/assets/50630549/d3f6703c-d17f-47bf-aefe-cd5788c632e3)

【事象2】
左デバイスで、問題のキーの赤点との穴（はんだ付けしていない部分）を通電すると「Key: R3 C2 Ke2」がOLEDに表示され、PCには"ecd"が入力されました。
![IMG_1703](https://github.com/Yowkees/keyball/assets/50630549/b83302a7-2518-4509-818c-a68fde411203)

**Comment by Ryo624** [comment]
導通ですが、テスターを使って確認をお願いいたします。
こちらの方のサイトに詳しく掲載されていますので、ご確認ください。
https://zenn.dev/halt/articles/20231029_keyball44

**Reply by y-y-yossy** [reply]
@Ryo624
テスターを購入し、導通確認を実施いたしました。

結果としては、両方とも抵抗値が1（オープン状態？）を示したため、導通できていないと思われます。
解決策のご教示のほどよろしくお願いいたします。

**Reply by y-y-yossy** [reply]
@Ryo624 
お忙しいところ大変恐縮ですが、この後の対応策があれば、ご教示いただきたく存じます。
よろしくお願いいたします。
